edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentController.cs

```diff
@@ -180,11 +198,11 @@ async Task ManageDeployments(V1DeploymentList existing, IEnumerable<V1Deployment
                     name =>
                     {
                         Events.DeleteDeployment(name);
-                        return this.client.DeleteNamespacedDeployment1Async(
+                        return this.client.DeleteNamespacedDeploymentAsync(
                             name,
                             this.deviceNamespace,
-                            propagationPolicy: "Foreground",
-                            body: new V1DeleteOptions(propagationPolicy: "Foreground"));
+                            propagationPolicy: KubernetesConstants.DefaultPropagationPolicy,
```

Comment:
Are we changing all unsuccessful deployment to be deleted in the background? If so, can you help understand why?

Reply:
These would be deployments being deleted because the module was removed from the Edge Deployment, or the entire Edge Deployment .  When we delete the module we're considering switching from "Foreground" delete propagation policy to "Background" for Deployment deletions.  Both should have the same net effect, but with background deletion, the API should respond more quickly.

We were a bit concerned that if the API call times out (modules with volumes seems to take a long time to delete), we might leave orphaned objects.  Also, we aren't waiting as long on deployments, so the operation takes less time.